test/python/primitives/test_statevector_sampler.py

```diff
@@ -0,0 +1,551 @@
+            self.assertEqual(len(result), 2)
+            self.assertEqual(result[0].data.meas.num_shots, shots1)
+            self.assertEqual(sum(result[0].data.meas.get_counts().values()), shots1)
+            self.assertIn("shots", result[0].metadata)
+            self.assertEqual(result[0].metadata["shots"], shots1)
+
+            self.assertEqual(result[1].data.meas.num_shots, shots2)
+            self.assertEqual(sum(result[1].data.meas.get_counts().values()), shots2)
+            self.assertIn("shots", result[1].metadata)
+            self.assertEqual(result[1].metadata["shots"], shots2)
+
+    def test_run_shots_result_size(self):
+        """test with shots option to validate the result size"""
+        n = 10
+        qc = QuantumCircuit(n)
+        qc.h(range(n))
+        qc.measure_all()
+        sampler = Sampler(seed=self._seed)
+        result = sampler.run([qc], shots=self._shots).result()
+        self.assertEqual(len(result), 1)
+        self.assertLessEqual(result[0].data.meas.num_shots, self._shots)
+        self.assertEqual(sum(result[0].data.meas.get_counts().values()), self._shots)
+
+    def test_primitive_job_status_done(self):
+        """test primitive job's status"""
+        bell, _, _ = self._cases[1]
+        sampler = Sampler(seed=self._seed)
+        job = sampler.run([bell], shots=self._shots)
+        _ = job.result()
+        self.assertEqual(job.status(), JobStatus.DONE)
+
+    def test_seed(self):
+        """Test for seed options"""
+        with self.subTest("empty"):
+            sampler = Sampler()
+            self.assertIsNone(sampler.seed)
+        with self.subTest("set int"):
+            sampler = Sampler(seed=self._seed)
+            self.assertEqual(sampler.seed, self._seed)
+        with self.subTest("set generator"):
+            sampler = Sampler(seed=np.random.default_rng(self._seed))
+            self.assertIsInstance(sampler.seed, np.random.Generator)
+
+    def test_circuit_with_unitary(self):
+        """Test for circuit with unitary gate."""
+
+        with self.subTest("identity"):
+            gate = UnitaryGate(np.eye(2))
+
+            circuit = QuantumCircuit(1)
+            circuit.append(gate, [0])
+            circuit.measure_all()
+
+            sampler = Sampler(seed=self._seed)
+            result = sampler.run([circuit], shots=self._shots).result()
+            self.assertEqual(len(result), 1)
+            self._assert_allclose(result[0].data.meas, np.array({0: self._shots}))
+
+        with self.subTest("X"):
+            gate = UnitaryGate([[0, 1], [1, 0]])
+
+            circuit = QuantumCircuit(1)
+            circuit.append(gate, [0])
+            circuit.measure_all()
+
+            sampler = Sampler(seed=self._seed)
+            result = sampler.run([circuit], shots=self._shots).result()
+            self.assertEqual(len(result), 1)
+            self._assert_allclose(result[0].data.meas, np.array({1: self._shots}))
+
+    def test_circuit_with_multiple_cregs(self):
```

Comment:
It might be good to also add a test where there are aliased registers, which can happen if you compose a circuit with conditionals into another circuit with a flat register. Eg for a bell teleportation like one

```python
q = QuantumRegister(3, 'q')
c1 = ClassicalRegister(1, 'c1')
c2 = ClassicalRegister(1, 'c2')

qc = QuantumCircuit(q, c1, c2)
qc.h(2)
qc.cx(2, 1)
qc.cx(0, 1)
qc.h(0)
qc.measure(0, c1)
qc.measure(1, c2)
qc.z(2).c_if(c1, 1)
qc.x(2).c_if(c2, 1)
qc.draw()


qc2 = QuantumCircuit(5, 5)
qc2.compose(qc, [0, 2, 3], [2, 4], inplace=True)
qc2.cregs
```
qc2 will have cregs 
```
[ClassicalRegister(5, 'c'),
 ClassicalRegister(1, 'c4'),
 ClassicalRegister(1, 'c5')]
```

Reply:
I found that register alias renamed register names ('c1' -> 'c4', 'c2' -> 'c5 in your example).
Is it possible to keep the register names?

Here is another case.
```python
from qiskit import QuantumCircuit, QuantumRegister, ClassicalRegister

q = QuantumRegister(3, 'q')
c1 = ClassicalRegister(1, 'c123456')
c2 = ClassicalRegister(1, 'ccccccc')

qc = QuantumCircuit(q, c1, c2)
qc.h(2)
qc.cx(2, 1)
qc.cx(0, 1)
qc.h(0)
qc.measure(0, c1)
qc.measure(1, c2)
qc.z(2).c_if(c1, 1)
qc.x(2).c_if(c2, 1)
print(qc.cregs)
# [ClassicalRegister(1, 'c123456'), ClassicalRegister(1, 'ccccccc')]
qc2 = QuantumCircuit(5, 5)
qc2.compose(qc, [0, 2, 3], [2, 4], inplace=True)
print(qc2.cregs)
# [ClassicalRegister(5, 'c'), ClassicalRegister(1, 'c0'), ClassicalRegister(1, 'c1')]
```

Reply:
Anyways, I added aliased registers to databin though they are renamed.